src/target/efm32.c

```diff
@@ -766,107 +711,127 @@ static bool efm32_mass_erase(target *t)
 		target_print_progress(&timeout);
 	}
 
+	/* Parts with >= 512 kiB flash have 2 mass erase regions */
+	if (flash_kib >= 512) {
+		/* Erase operation */
+		target_mem_write32(t, EFM32_MSC_WRITECMD(msc), EFM32_MSC_WRITECMD_ERASEMAIN1);
+
+		/* Poll MSC Busy */
+		while ((target_mem_read32(t, EFM32_MSC_STATUS(msc)) & EFM32_MSC_STATUS_BUSY)) {
+			if (target_check_error(t))
+				return false;
+		}
```

Comment:
This needs a call to `target_print_progress(&timeout);` inside the loop

Reply:
oops, just blindly rebased